src/hooks/useAutoSlippageTolerance.ts

```diff
@@ -100,9 +99,7 @@ export default function useAutoSlippageTolerance(
     // NOTE - dont use gas estimate for L2s yet - need to verify accuracy
     // if not, use local heuristic
     const dollarCostToUse =
-      chainId && SUPPORTED_GAS_ESTIMATE_CHAIN_IDS.includes(chainId) && trade?.gasUseEstimateUSD
-        ? trade.gasUseEstimateUSD
-        : dollarGasCost
+      chainId && SUPPORTED_GAS_ESTIMATE_CHAIN_IDS.includes(chainId) && gasEstimateUSD ? gasEstimateUSD : dollarGasCost
```

Comment:
This is one thing that makes me often confused. If the trade is calculated against a particular chainId (say I am swapping on Polygon), then I would expect `gasEstimateUSD` to be null if the chain does not support it. 

My point is, I find `chainId && SUPPORTED_GAS_ESTIMATE_CHAIN_IDS.includes(chainId)` unnecessary and feel like it would be great to do where we create `trade` object. That way, we could remove A LOT of conditionals from the codebase, I have seen that one around in particular.

Reply:
I agree - why can't this check just be `trade?.gasEstimateUSD ? gasEstimateUSD : dollarGasCost`

Reply:
this feels outside of the scope of this PR